src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/AbstractDocumentService.java

```diff
@@ -88,4 +108,63 @@ CaseDetails copyOf(CaseDetails caseDetails) {
             throw new IllegalStateException();
         }
     }
+
+    CaseDetails prepareNotificationLetter(CaseDetails caseDetails) {
+        Map<String, Object> caseData = caseDetails.getData();
+        Map addressToSendTo;
+
+        String ccdNumber = nullToEmpty((caseDetails.getId()));
+        String reference = "";
+        String addresseeName;
+        String applicantName = nullToEmpty((caseData.get(APPLICANT_FIRST_MIDDLE_NAME)))
+            + " " + nullToEmpty((caseData.get(APPLICANT_LAST_NAME)));
```

Comment:
I think we don't want this blank space if the first name is not present. I think an easier way to achieve this is by using Optional.